control-plane/agents/src/bin/core/controller/io_engine/v1/translation.rs

```diff
@@ -127,6 +127,114 @@ impl IoEngineToAgent for v1::replica::ReplicaSpaceUsage {
     }
 }
 
+impl TryIoEngineToAgent for v1::nexus::NexusCreateSnapshotResponse {
+    type AgentMessage = transport::CreateNexusSnapshotResp;
+    fn try_to_agent(&self) -> Result<Self::AgentMessage, SvcError> {
+        // Extract uuid string locally for convenience.
+        let uuid_str = if let Some(u) = self.nexus.as_ref().map(|n| n.uuid.clone()) {
+            u
+        } else {
+            return Err(SvcError::InvalidArguments {});
+        };
+        Ok(Self::AgentMessage {
+            nexus: NexusId::try_from(uuid_str.as_str()).map_err(|_| SvcError::InvalidUuid {
+                uuid: uuid_str.to_owned(),
+                kind: ResourceKind::Nexus,
+            })?,
+            snap_time: self
+                .snapshot_timestamp
+                .clone()
+                .and_then(|t| std::time::SystemTime::try_from(t).ok())
+                .unwrap_or(UNIX_EPOCH),
+            replicas_status: self
+                .replicas_done
+                .iter()
+                .map(|s| s.try_to_agent())
+                .collect::<Result<_, _>>()?,
+            skipped: self
+                .replicas_skipped
+                .iter()
+                .map(|s| {
+                    ReplicaId::try_from(s.as_str()).map_err(|_| SvcError::InvalidUuid {
+                        uuid: s.to_owned(),
+                        kind: ResourceKind::Replica,
+                    })
+                })
+                .collect::<Result<_, _>>()?,
+        })
+    }
+}
+
+/// Helper implementation to be used as part of nexus snapshot
+/// response translation.
+impl TryIoEngineToAgent for v1::nexus::NexusCreateSnapshotReplicaStatus {
+    type AgentMessage = transport::CreateNexusSnapshotReplicaStatus;
+    fn try_to_agent(&self) -> Result<Self::AgentMessage, SvcError> {
+        Ok(Self::AgentMessage {
+            replica_uuid: ReplicaId::try_from(self.replica_uuid.as_str()).map_err(|_| {
+                SvcError::InvalidUuid {
+                    uuid: self.replica_uuid.to_owned(),
+                    kind: ResourceKind::Replica,
+                }
+            })?,
+            status: self.status_code,
+        })
+    }
+}
+
+impl TryIoEngineToAgent for v1::replica::CreateReplicaSnapshotResponse {
+    type AgentMessage = transport::CreateReplicaSnapshotResp;
+    fn try_to_agent(&self) -> Result<Self::AgentMessage, SvcError> {
+        let descriptor = if let Some(d) = self.snapshot.as_ref() {
+            d.try_to_agent()?
+        } else {
+            return Err(SvcError::InvalidArguments {});
+        };
+        Ok(Self::AgentMessage {
+            replica: ReplicaId::try_from(self.replica_uuid.clone()).map_err(|_| {
```

Comment:
this should be replica rather than ReplicaId

Reply:
ah, @hrudaya21 this comment was resolved but not done!
https://github.com/openebs/mayastor-api/pull/56#discussion_r1191378354

Reply:
> ah, @hrudaya21 this comment was resolved but not done! [openebs/mayastor-api#56 (comment)](https://github.com/openebs/mayastor-api/pull/56#discussion_r1191378354)

It was updated as below, part of review comment in that PR @tiagolobocastro 
/// Create Replica Snapshot Response.
message CreateReplicaSnapshotResponse {
  string replica_uuid = 1; // UUID of the source replica
  **ReplicaSnapshot snapshot = 2; // Snapshot detail information**
}

Reply:
Ah sorry I missed the comment meant, I thought there was another to change 
string replica_uuid = 1;
to
Replica replica = 1;

Reply:
Alternatively, if replica itself does not change what we'd return over grpc, then we can simply omit this entirely..